### PR TITLE
Human vs computer

### PR DESCRIPTION
**Pull request by Himalee**

### CODE REVIEW

File: lib/computer_player.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    randomise
+    display_input
+  end
+
+  def randomise
+    options = ["rock", "paper", "scissors"]
+    @player_input = options.sample
+  end
+
+  def display_input
```
Comment (fabientownsend) [inline lib/computer_player.rb]: Is that really display an input correct in this case? as a value _inserted_
Wouldn't something like display_shape more accurate?

File: lib/computer_player.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  def get_input
+    @display.present(Messages.new.player_prompt(@name))
+    randomise
+    display_input
+  end
+
+  def randomise
+    options = ["rock", "paper", "scissors"]
+    @player_input = options.sample
```
Comment (fabientownsend) [inline lib/computer_player.rb]: I think you could inline that as:
```
  @player_input = random_shape
end
def random_shape
  ["rock", "paper", "scissors"].sample
end

File: lib/game.rb [hunk 3]
```diff
@@ -12,27 +13,37 @@ def initialize(player_one, player_two, display, messages)
...
-    @winning_option = rules.outcome
+  def option_one_result(option_one, option_two)
+    turn = [option_one, option_two]
+    if option_one == option_two
+      "draw"
+    elsif turn.include?("rock") and turn.include?("scissors") and option_one == "rock"
+      "win"
+    elsif turn.include?("paper") and turn.include?("rock") and option_one == "paper"
+      "win"
+    elsif turn.include?("paper") and turn.include?("scissors") and option_one == "scissors"
```
Comment (fabientownsend) [inline lib/game.rb]: It feels like that you could refactor here with https://sourcemaking.com/refactoring/extract-method

`elsif winner?` (this is a just an example, I'm sure you can find a better name)

Also, the name `option` is quite vague, do you think there is a more descriptive name that you could use?

File: lib/computer_player.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+require_relative "player"
+
+class ComputerPlayer < Player
+
+  attr_reader :player_input
```
Comment (devlinglasman) [inline lib/computer_player.rb]: This line is in both player classes - is it possible therefore to elevate it to superclass level?

File: lib/computer_player.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  def get_input
+    @display.present(Messages.new.player_prompt(@name))
+    get_shape
+    display_shape
+  end
+
+  def random_shape
```
Comment (devlinglasman) [inline lib/computer_player.rb]: I don't think they're really 'shapes' - can you think of a better name?

File: lib/computer_player.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+require_relative "player"
+
+class ComputerPlayer < Player
+
+  attr_reader :player_input
+
+  def get_input
+    @display.present(Messages.new.player_prompt(@name))
```
Comment (devlinglasman) [inline lib/computer_player.rb]: This is also in both subclasses. You should look into what the keyword 'super' does.

File: lib/player.rb [hunk 7]
```diff
@@ -18,9 +18,4 @@ def paper
   def scissors
```
Comment (devlinglasman) [inline lib/player.rb]: I'm not sure why this is here?

File: lib/game.rb [hunk 8]
```diff
@@ -12,27 +13,37 @@ def initialize(player_one, player_two, display, messages)
...
   end
 
-  def determine_winner
-    rules = Rules.new(@player_one.user_input, @player_two.user_input)
-    @winning_option = rules.outcome
+  def option_one_result(option_one, option_two)
+    turn = [option_one, option_two]
+    if option_one == option_two
+      "draw"
+    elsif turn.include?("rock") and turn.include?("scissors") and option_one == "rock"
```
Comment (devlinglasman) [inline lib/game.rb]: You're checking at the end of this statement that Option One must be 'rock' - then you're asking that a 'rock' and a 'scissors' are included in the two-item array, which is the only situation in which this statement returns true... can you see duplication/ problems with that?

**Pull request merged**